course4/homework-list/test.cc

```diff
@@ -0,0 +1,47 @@
+#include <iostream>
+#include "list.hh"
+#include <cassert>
+#include <algorithm>
+#include<forward_list>
+
+template<typename T>
+void RunStringTest()
+{
+    T list;
+    list.push_front("ABC");
+    assert(!list.empty());
+    list.push_front("XYZ");
+    assert(list.front() == "XYZ");
+    list.pop_front();
+    assert(list.front() == "ABC");
+    list.pop_front();
+    assert(list.empty());
+    list.push_front("OOO");
+    assert(list.front() == "OOO");
+    list.push_front("XYZ");
+    assert(list.front() == "XYZ");
+}
+
+template<typename T>
+void RunAlgoTest()
+{
+    T list;
+    list.push_front(2);
+    list.push_front(3);
+    list.push_front(2);
+    list.push_front(1);
+    list.push_front(2);
+
+    int count = std::count_if(list.begin(), list.end(), [](int x) {return x == 2;});
+    assert(count == 3);
+}
+
```

Comment:
Please, wrtie the followign algorithm, and thest both containers on it:

"count_until" - an algorithm that counts the occurences of a value, nween the beginning of a range, and the first occurence of some "stop" element.
If there is no "stop" element, then the count should encompass the entire container.

Examples:
````
list = {1, 2, 1, ,2, 3, 1, ,2 };
auto count1 = count_until(list.begin(), list.end(), 1/*search*/, 3/*stop*/);
assert(count1 == 2);
auto count2 = count_until(list.begin(), list.end(), 1/*search*/, 7/*stop*/);
assert(count2 == 3);
```